src/Slackbot.Net.Extensions.FplBot/Abstractions/ISlackWorkSpacePublisher.cs

```diff
@@ -1,11 +1,13 @@
+using Slackbot.Net.SlackClients.Http.Models.Responses.UsersList;
+using System;
 using System.Threading.Tasks;
 
 namespace Slackbot.Net.Extensions.FplBot.GameweekLifecycle
 {
     public interface ISlackWorkSpacePublisher
     {
-        Task PublishToAllWorkspaces(string msg);
-        Task PublishUsingToken(string token, params string[] messages);
-        Task PublishToSingleWorkspaceConnectedToLeague(int leagueId, params string[] messages);
+        Task PublishToAllWorkspaces(Func<User[], string> msg);
+        Task PublishUsingToken(string token, params Func<User[], string>[] messages);
+        Task PublishToSingleWorkspaceConnectedToLeague(int leagueId, params Func<User[], string>[] messages);
     }
```

Comment:
I was thinking more that this interface would be kept as-is just for publishing, and that the messages and formatting would be done before calling this (`messages` would have the slack  handles taunts when calling these methods).